networkml/algorithms/host_footprint_predict.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pandas as pd
+import json
+
+def predict(input_path, model, label_encoder, scaler_fitted):
+    """
+    This function takes a csv of features at the host footprint level and then makes
+    a role prediction for each row. The output is the top three roles.
+
+    INPUTS:
+    --input_path: the location of the .csv with host footprint features (correct assumption?)
+    --model: a trained model
+    --label_encoder: a mapping of numeric ouput to roles?
+    
+    OUTPUTS:
+    --prediction: top three roles for each host and the associated probability of each role
+    (in json format)
+    """
+    
+    ## Load data from host footprint .csv
+    df = pd.read_csv(input_path)
+    
+    ## Check if there is greater than 1 row, there should
+    ## only be one row
+    if df.shape[0] > 1:
+        
+        pass
+        ## log warning
+        ## TODO: JOHN SPEED NEEDS HELP USING THE PROPER SYNTAX
+        #logging.error(f'More than one row found in predict_input csv')      
+    
+    ## Split dataframe into X (the input features or predictors)
+    ## and y (the target or outcome or dependent variable)
+    ## This drop function should work even if there is no column
+    ## named filename
+    X = df.drop("filename", axis=1)
+
+    ## Normalize X features before training
+    scaler = preprocessing.MinMaxScaler()
```

Comment:
preprocessing needs to be imported

Reply:
Got it. I will update the PR.

Reply:
you have to update the call with the way you imported that, or update the way you're importing